### DISCUSSION TITLE
Accessing mode spectra and central frequencies in VmdTransformer

### DISCUSSION THREAD

**Discussion by anates** [alan-turing-institute/sktime#6745] [General]
Previously, I used vmdpy for applying mode decompositions to my signal. There, it was possible to retrieve not only the modes, but also the mode spectra and central frequencies of the modes. 

Now I switched to the vmd-implementation in sktime. However, when calling fit_transform(), I only get the modes, but nothing else. How can I access the additional two parameters (if at all)?

Thanks!

**Comment by fkiraly** [comment]
* you can get `vmdpy` still from `sktime`, it is in `libs.vmdpy`, so a prior import `from vmdpy import stuff` can be replaced by `from sktime.libs.vmdpy import stuff`, all your prior code should still work unchanged after you change the imports
    * the transformation is useful for use in `sktime` pipelines, for direct plug & play
* I assume you mean `u_hat`, `omega`, right? These seem to be discarded internally, but in-principle these could be returned (controlled by a parameter added), or stored in `self.omega_` etc, then you could get them via `get_fitted_params`
    * I would advise to open an issue with a feature request! In that issue, it would also be great if you could share your thoughts on whether these should be returned as fitted parameters, or whether the user should be able to choose what `transform` should return.
